### PR TITLE
Python: add LINSERT command

### PR DESCRIPTION
**Pull request by aaron-congo**
https://redis.io/docs/latest/commands/linsert/

### CODE REVIEW

File: python/python/glide/async_commands/core.py [hunk 1]
```diff
@@ -184,6 +184,11 @@ def get_cmd_args(self) -> List[str]:
         return [self.cmd_arg] if self.value is None else [self.cmd_arg, self.value]
 
 
+class InsertPosition(Enum):
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/core.py:187]: Is it auto-exported to a user?

File: python/python/tests/test_transaction.py [hunk 2]
```diff
@@ -32,6 +33,7 @@ async def transaction_test(
     key7 = "{{{}}}:{}".format(keyslot, get_random_string(3))
     key8 = "{{{}}}:{}".format(keyslot, get_random_string(3))
     key9 = "{{{}}}:{}".format(keyslot, get_random_string(3))
+    key10 = "{{{}}}:{}".format(keyslot, get_random_string(3))
```
Comment (Yury-Fridlyand) [inline python/python/tests/test_transaction.py:36]: maybe add a hint for that key
```suggestion
    key10 = "{{{}}}:{}".format(keyslot, get_random_string(3)) # list
```

File: python/python/tests/test_async_client.py [hunk 3]
```diff
@@ -907,6 +908,34 @@ async def test_rpushx(self, redis_client: TRedisClient):
...
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_linsert(self, redis_client: TRedisClient):
+        key1 = get_random_string(10)
+        key2 = get_random_string(10)
+
+        assert await redis_client.lpush(key1, ["4", "3", "2", "1"]) == 4
+        assert await redis_client.linsert(key1, InsertPosition.BEFORE, "2", "1.5") == 5
+        assert await redis_client.linsert(key1, InsertPosition.AFTER, "3", "3.5") == 6
+        assert await redis_client.lrange(key1, 0, -1) == [
+            "1",
```
Comment (Yury-Fridlyand) [inline python/python/tests/test_async_client.py:921]: Don't trust person whom you stole that test from!

File: python/python/glide/async_commands/transaction.py [hunk 4]
```diff
@@ -771,6 +772,28 @@ def rpop_count(self: TTransaction, key: str, count: int) -> TTransaction:
         """
         return self.append_command(RequestType.RPop, [key, str(count)])
 
+    def linsert(
+        self: TTransaction, key: str, position: InsertPosition, pivot: str, element: str
+    ) -> TTransaction:
+        """
+        Inserts `element` in the list at `key` either before or after the `pivot`.
+        See https://redis.io/commands/linsert/ for details.
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/transaction.py:780]: I think we need a blank line before `see` and `command response`

File: python/python/glide/async_commands/core.py [hunk 5]
```diff
@@ -1042,6 +1047,35 @@ async def rpop_count(self, key: str, count: int) -> Optional[List[str]]:
             await self._execute_command(RequestType.RPop, [key, str(count)]),
         )
 
+    async def linsert(
+        self, key: str, position: InsertPosition, pivot: str, element: str
+    ) -> int:
+        """
+        Inserts `element` in the list at `key` either before or after the `pivot`.
+        See https://redis.io/commands/linsert/ for details.
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/core.py:1055]: I think we need a blank line before see and returns

File: python/python/glide/async_commands/core.py (hunk 1 above)
Comment (aaron-congo) [reply python/python/glide/async_commands/core.py:187]: Users will have to import it, like they would for the other enums in this file as well (eg ExpireOptions).

File: python/python/tests/test_async_client.py (hunk 3 above)
Comment (aaron-congo) [reply python/python/tests/test_async_client.py:921]: Can't tell if you're joking haha. Took it from your PR here:
https://github.com/Bit-Quill/glide-for-redis/pull/176/files#diff-5db9a3bd41d647cfdba521ea248ea07d7c78bc9aaa41a26bb923d4f92a3efdd2R1223

**Review by SanHalacogluImproving — APPROVED** [review]

File: python/python/tests/test_async_client.py (hunk 3 above)
Comment (Yury-Fridlyand) [reply python/python/tests/test_async_client.py:921]: I know, don't trust me

File: python/python/glide/async_commands/core.py [hunk 6]
```diff
@@ -1042,6 +1047,36 @@ async def rpop_count(self, key: str, count: int) -> Optional[List[str]]:
...
+
+        See https://redis.io/commands/linsert/ for details.
+
+        Args:
+            key (str): The key of the list.
+            position (InsertPosition): The relative position to insert into - either `InsertPosition.BEFORE` or
+                `InsertPosition.AFTER` the `pivot`.
+            pivot (str): An element of the list.
+            element (str): The new element to insert.
+        Returns:
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/core.py:1064]: newlike

File: python/python/glide/async_commands/transaction.py [hunk 7]
```diff
@@ -771,6 +772,29 @@ def rpop_count(self: TTransaction, key: str, count: int) -> TTransaction:
...
+
+        See https://redis.io/commands/linsert/ for details.
+
+        Args:
+            key (str): The key of the list.
+            position (InsertPosition): The relative position to insert into - either `InsertPosition.BEFORE` or
+                `InsertPosition.AFTER` the `pivot`.
+            pivot (str): An element of the list.
+            element (str): The new element to insert.
+        Command response:
```
Comment (Yury-Fridlyand) [inline python/python/glide/async_commands/transaction.py:789]: newline

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**